core/sdk-core/src/main/java/software/amazon/awssdk/core/Response.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+
+/**
+ * Response wrapper that indicates success or failure with the associated unmarshalled response object or exception
+ * object. This object is used by the core request/response pipeline to pass response metadata alongside the actual
+ * deserialized response object through different stages of the pipeline.
+ *
+ * @param <T> the modelled SDK response type.
+ */
+@SdkProtectedApi
```

Comment:
internal api?

Reply:
It's now protected because the codegen has responsibility to pass a combined handler that produces a Response object based on protocol. That means there's now a direct reference to this class from generated code.